Code/src/main/java/nl/utwente/group10/ui/BackendUtils.java

```diff
@@ -31,4 +33,18 @@ public static Type dive(ConstT function, int steps) {
         }
         return function;
     }
+    
+    /**
+     * @return Whether or not t1 and t2 unify with each other.
```

Comment:
I have asked previously how to check if types match, and to my knowledge have not received a consise answer, so I did it this way. But again, this PR is not intended to change the type system and the way it is used, as @TheMSB mentioned, this PR is big. Changing all the type stuff would make it bigger.

For the future: I dont care how it happens, I just need a method that can tell me if 2 types match.